docs/proxies.rst

```diff
@@ -71,3 +71,12 @@ When you implement proxies, the best documentation to follow is the source of To
 
 Sometimes precondition checks are impossible because the specified block is too old (pruned in the Ethereum client).
 In this case, the precondition check can be skipped.
+
+When you implement a new method of a proxy, consider refusing the string ``"last"`` as the block identifier.
+Usually passing ``"last"`` as a block identifier to a proxy method poses a possibility of ``BrokenPreconditionError``.
+The method can raise a ``ValueError`` when ``"last"`` is passed as the block identifier.
+Or, you might want to introduce a new type so that ``mypy`` complains against using ``"last"`` as the block identifier.
```

Comment:
I would prefer adding the type instead of the documentation ... there is really no guarantee that people will read this doc, and this is not machine verifiable. In contrast adding a type to the code that only works on the aforementioned tags will be enforced by the build

Reply:
Here is an issue that keeps track of doing that. https://github.com/raiden-network/raiden/issues/4949